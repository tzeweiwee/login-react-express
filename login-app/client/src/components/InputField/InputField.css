.inputContainer {
  width: 100%;
}

.labelText {
  width: 30%;
  font-size: 16px;
  margin: 5px;
  float:left;
}

.inputField {
  width: 60%;
  margin: 5px;
}