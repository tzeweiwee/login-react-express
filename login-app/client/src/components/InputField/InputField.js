import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

export default class InputField extends PureComponent {
    render () {
        const { name, labelText, value, onChange, type, placeholder } = this.props;
        console.log(this.props);
        return (
            <div className='inputContainer'>
                <label className='labelText'>{labelText}</label>
                <input name={name} type={type} className='inputField' placeholder={placeholder} value={value} onChange={onChange} />
            </div>
        );
    }
};

InputField.defaultProps = {
  type: "text"
};

InputField.propTypes = {
    onChange: PropTypes.func.isRequired,
    labelText: PropTypes.string.isRequired,
    name: PropTypes.string,
    value: PropTypes.string,
    type: PropTypes.string,
    placeholder: PropTypes.string
};