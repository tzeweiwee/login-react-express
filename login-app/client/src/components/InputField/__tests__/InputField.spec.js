import React from 'react';
import { shallow } from 'enzyme';
import InputField from '../InputField';

const inputProps = {
  labelText: "label",
  type:"text",
  value: '',
  onChange: () => { },
  placeholder: 'placeholder'
}

describe('InputField', () => {
  it('should render a textbox component', () => {
      const wrapper = shallow(<InputField {...inputProps}/>);
      expect(wrapper).toHaveLength(1);
  });

  it('should match InputField snapshot', () => {
    const wrapper = shallow(<InputField {...inputProps}/>);
    expect(wrapper).toMatchSnapshot();
});
});
