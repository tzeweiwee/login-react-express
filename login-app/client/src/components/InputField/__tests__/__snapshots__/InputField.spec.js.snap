// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should match InputField snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    labelText="label"
    onChange={[Function]}
    placeholder="placeholder"
    type="text"
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className={undefined}
>
          label
          :
</label>,
        <input
          className={undefined}
          name={undefined}
          onChange={[Function]}
          placeholder="placeholder"
          type="text"
          value=""
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "label",
            ":",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          "label",
          ":",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "name": undefined,
          "onChange": [Function],
          "placeholder": "placeholder",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className={undefined}
>
            label
            :
</label>,
          <input
            className={undefined}
            name={undefined}
            onChange={[Function]}
            placeholder="placeholder"
            type="text"
            value=""
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "label",
              ":",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "label",
            ":",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "name": undefined,
            "onChange": [Function],
            "placeholder": "placeholder",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
